tests/test_views.py

```diff
@@ -0,0 +1,455 @@
+            manufacturer=self.manufacturer
+        )
+        self.car_data = {
+            "model": "New Model",
+            "manufacturer": self.manufacturer_2.id,
+            "drivers": []
+        }
+
+        self.all_cars = Car.objects.all()
+        self.client.force_login(self.user)
+
+    def test_logged_in_user_access(self) -> None:
+        for url in (
+                CAR_LIST_URL,
+                CAR_CREATE_URL,
+                get_car_detail_url(self.car),
+                get_car_update_url(self.car),
+                get_car_delete_url(self.car)
+        ):
+            http_response = self.client.get(url)
+            self.assertEqual(
+                http_response.status_code,
+                200
+            )
+
+    def test_list(self) -> None:
+        http_response = self.client.get(CAR_LIST_URL)
+        self.assertEqual(
+            list(http_response.context["car_list"]),
+            list(self.all_cars)
+        )
+
+    def test_list_search(self) -> None:
+        for search_letter in ("5", "6", "m"):
+            http_response = self.client.get(
+                CAR_LIST_URL,
+                {"model": f"{search_letter}"}
+            )
+            self.assertEqual(
+                list(http_response.context["car_list"]),
+                list(self.all_cars.filter(model__icontains=f"{search_letter}"))
+            )
+
+    def test_detail(self) -> None:
+        self.assertEqual(
+            self.client.get(get_car_detail_url(self.car)).context["car"],
+            self.car
+        )
+        self.assertEqual(
+            self.client.get(get_car_detail_url(self.car_2)).context["car"],
+            self.car_2
+        )
+
+    def test_create(self) -> None:
+        http_response = self.client.post(
+            CAR_CREATE_URL,
+            data=self.car_data
+        )
+        car = Car.objects.get(
+            model=self.car_data["model"]
+        )
+        self.assertEqual(
+            http_response.status_code,
+            302
+        )
+        self.assertEqual(
+            car.manufacturer,
+            self.manufacturer_2
+        )
+
+    def test_update(self) -> None:
+        http_response = self.client.post(
+            get_car_update_url(self.car),
+            data=self.car_data
+        )
+        car = Car.objects.get(
+            model=self.car_data["model"]
+        )
+        self.assertEqual(
+            http_response.status_code,
+            302
+        )
+        self.assertEqual(
+            car.manufacturer,
+            self.manufacturer_2
+        )
+
+    def test_delete(self) -> None:
+        http_response = self.client.post(get_car_delete_url(self.car))
+        http_response_2 = self.client.post(get_car_delete_url(self.car_2))
+        self.assertEqual(
+            http_response.status_code,
+            302
+        )
+        self.assertEqual(
+            http_response_2.status_code,
+            302
+        )
+
+        self.assertFalse(len(self.all_cars))
```

Comment:
In the `test_delete` method of `PrivateCarTest`, consider verifying that the car objects are actually deleted from the database by checking that they cannot be retrieved anymore, rather than just checking the length of the queryset.